lib/mix/tasks/phx.gen.auth.ex

```diff
@@ -285,6 +302,12 @@ defmodule Mix.Tasks.Phx.Gen.Auth do
     |> Kernel.not()
   end
 
+  defp generated_with_no_assets_or_esbuild? do
+    not File.exists?("assets/js/app.js") or
+      File.read!("assets/js/app.js") =~
+        "For Phoenix.HTML support, including form and button helpers"
+  end
```

Comment:
I am afraid this will have false positives if someone chooses to have their assets elsewhere. Also asserting on the text may be brittle.

My suggestion would be:

```suggestion
  defp generated_with_no_assets_or_esbuild? do
    not Code.ensure_loaded?(Phoenix.HTML) or
      case File.read("assets/js/app.js") do
        {:ok, content} -> content =~ "priv/static/phoenix_html.js"
        {:error, _} -> true
      end
  end
```

Although I think this means we won't catch the --no-assets case... but then I am not sure if there is much we can do.

Reply:
I think we can check the case where we generate the file without imports but that’s it.